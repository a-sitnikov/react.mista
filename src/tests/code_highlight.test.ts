import highlight from "src/components/extensions/code_highlight";

describe("Code highlight", () => {
  it("Процедура с комментарием", () => {
    const text = highlight(
      "// Комментарий\n" +
        "Процедура Тест()\n" +
        "  а = 1;\n" +
        "КонецПроцедуры\n"
    );

    expect(text).toMatchSnapshot();
  });

  it("Многострочный литерал", () => {
    const text = highlight('d = "\n' + "| dfsf\n" + '| dsgffsg";');

    expect(text).toMatchSnapshot();
  });
});
