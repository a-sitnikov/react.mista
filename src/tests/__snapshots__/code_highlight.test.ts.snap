// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code highlight test 1 1`] = `
"<span class=com>// Комментарий</span>
<span class=key>Процедура </span><span class=bg>Тест</span><span class=sp>()</span>
  <span class=bg>а </span><span class=sp>= </span><span class=num>1</span><span class=sp>;</span>
<span class=key>КонецПроцедуры</span>
"
`;

exports[`Code highlight test 2 1`] = `
"<span class=bg>d </span><span class=sp>= </span><span class=str>\\"
| dfsf
| dsgffsg\\"</span><span class=sp>;</span>"
`;
